src/PKSim.Core/Services/DistributedParameterToTableParameterConverter.cs

```diff
@@ -155,26 +158,35 @@ private void createSpatialStructureTableParameters(IBuildConfiguration buildConf
-         foreach (var baseIndividualParameter in allBaseIndividualDistributedParameters.KeyValues)
+
+         foreach (var individualParameterKeyValue in allBaseIndividualDistributedParameters.KeyValues)
          {
-            var structureParameter = allContainerParameters[baseIndividualParameter.Key] ?? allNeighborhoodParameters[baseIndividualParameter.Key];
+            var individualParameterPath = individualParameterKeyValue.Key;
+            var individualParameter = individualParameterKeyValue.Value;
+            var structureParameter = allContainerParameters[individualParameterPath] ?? allNeighborhoodParameters[individualParameterPath];
             if (structureParameter == null)
                continue;
 
             //cache all distributions for this parameter defined for the population and sub population.
-            var allDistributionsForParameter = _parameterQuery.ParameterDistributionsFor(baseIndividualParameter.Value.ParentContainer, _baseOriginData.SpeciesPopulation, _baseOriginData.SubPopulation, baseIndividualParameter.Value.Name);
+            var allDistributionsForParameter = _parameterQuery.ParameterDistributionsFor(individualParameter.ParentContainer, _baseOriginData.SpeciesPopulation, _baseOriginData.SubPopulation, individualParameter.Name);
             var allDistributionsForMaleParameter = allDistributionsForParameter.Where(p => p.Gender == CoreConstants.Gender.Male).ToList();
             var allDistributionsForFemaleParameter = allDistributionsForParameter.Where(p => p.Gender == CoreConstants.Gender.Female).ToList();
 
-            createSpatialStructureTableParameter(structureParameter, baseIndividualParameter.Value, allDistributionsForMaleParameter, allDistributionsForFemaleParameter, buildConfiguration);
+            createSpatialStructureTableParameter(structureParameter, individualParameter, allDistributionsForMaleParameter, allDistributionsForFemaleParameter, buildConfiguration);
 
-            createPopulationTableParameter(baseIndividualParameter, simulationPopulation, allDistributionsForMaleParameter, allDistributionsForFemaleParameter);
+            createPopulationTableParameter(individualParameterPath, individualParameter, populationSimulation, allDistributionsForMaleParameter, allDistributionsForFemaleParameter);
          }
+
+         //Height parameter is not a distributed parameter. However, we need to define this parameter as table to ensure that height dependent parameters are updated properly
+         var heightParameter = spatialStructure.TopContainers.FindByName(Constants.ORGANISM).Parameter(CoreConstants.Parameters.HEIGHT);
+         var individualMeanHeightParameter = _baseIndividual.Organism.EntityAt<IDistributedParameter>(CoreConstants.Parameters.MEAN_HEIGHT);
+         createSpatialStructureTableParameter(heightParameter, individualMeanHeightParameter, _allHeightDistributionMaleParameters, _allHeightDistributionMaleParameters, buildConfiguration);
```

Comment:
@Yuri05 I am adding now a table parameter for the height parameter as well as an entry in the aging table

Reply:
createSpatialStructureTableParameter(heightParameter, individualMeanHeightParameter, _allHeightDistribution**Male**Parameters, _allHeightDistribution**Male**Parameters, buildConfiguration);

one of them should be female I guess

Reply:
good catch!

Reply:
fixed